### PR TITLE
adds click

### LINKED ISSUE #33: commandline: path to config
Add commandline functionality to specify a path to a config

```sh
blackbox --config=path/to/config.yml
```

and of course have a sensible default

### PR DESCRIPTION
**Pull request by Inveracity**
closes #33

### CODE REVIEW

**Comment by Inveracity** [comment]
woops, this should've been a draft PR

**Comment by Inveracity** [comment]
this one is ready to be looked

**Comment by Inveracity** [comment]
Apologies for all the formatting edits, VSCode seems to be quite opinionated on how to format dockerfiles and readmes

File: README.md [hunk 1]
```diff
@@ -7,21 +7,61 @@ A simple service which magically backs up all your databases to all your favorit
...
 # Setup
+
 This service can either be set up as a cron job (on UNIX systems), as a Kubernetes CronJob, or scheduled in your favorite alternative scheduler.
 
+## Quick start
+
+Requires Python 3.9 or newer
+
+```sh
+# TODO: decide on this naming convention
```
Comment (lemonsaurus) [inline README.md:18]:
```suggestion
# Install the CLI tool
```
`blackbox-cli` will do.

File: README.md [hunk 2]
```diff
@@ -7,21 +7,61 @@ A simple service which magically backs up all your databases to all your favorit
...
+    spec:
+      template:
+        spec:
+          containers:
+            - name: blackbox-backups
+              image: lemonsaurus/blackbox:latest
+              imagePullPolicy: Always
+              args:
+                - blackbox
+          restartPolicy: OnFailure
```
Comment (lemonsaurus) [inline README.md:60]:
```suggestion
```

I'll add this once we've verified it working for PyDis.

File: docker/Dockerfile [hunk 3]
```diff
@@ -32,4 +30,4 @@ RUN pipenv install --system --deploy
 COPY . .
 
 # Start the application!
-CMD ["python", "main.py"]
+CMD ["python", "-m", "blackbox"]
```
Comment (lemonsaurus) [inline docker/Dockerfile:33]:
```suggestion
CMD ["python", "-m", "blackbox"]
```
Maybe we actually use the `blackbox` command here, as a sanity check? Just gotta install it with pip install -e . first.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Minor changes here before we merge.

File: blackbox/config.py [hunk 4]
```diff
@@ -43,21 +43,23 @@ class Prefixes(metaclass=YAMLGetter):
     @classmethod
     def parse_config(cls, config_path: Path = None):
         """Parse the config from the blackbox.yaml file."""
+
```
Comment (lemonsaurus) [inline blackbox/config.py:46]:
```suggestion
```

Newline after docstring is **ILLEGAL** and ⛤⛧⛥**DEMONIC** ⛤⛧⛥

File: blackbox/config.py [hunk 5]
```diff
@@ -43,21 +43,23 @@ class Prefixes(metaclass=YAMLGetter):
...
         # If config_path is passed, use that.
+        env_config_path = os.environ.get("BLACKBOX_CONFIG_PATH")
+
+        # If there is a path to the config, parse it
```
Comment (lemonsaurus) [inline blackbox/config.py:49]: I'm thinking the comment on line 47 can just be above `if config_path`, where it seems to belong.

The environment get doesn't need a block comment.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]

File: blackbox/cli.py [hunk 6]
```diff
@@ -1,8 +1,53 @@
...
+        if not config_file.exists():
+            config_file.write_text(dedent(
+                """
+                databases:
+                  - mongodb://username:password@host:port
+                  - postgres://username:password@host:port
+                  - redis://password@host:port
+
+                storage:
+                  - s3://username:password?fire=ice&magic=blue
```
Comment (lemonsaurus) [inline blackbox/cli.py:33]:
```suggestion
                  - s3://bucket:s3.endpoint.com?aws_access_key_id=1234&aws_secret_access_key=lemondance
```

**Review by lemonsaurus — APPROVED** [review]
one minor change more.

**Pull request merged**